modules/extension/xsd/xsd-gml2/src/main/java/org/geotools/gml2/simple/GMLWriter.java

```diff
@@ -278,6 +316,10 @@ void position(double x, double y, double z, StringBuffer sb) {
             sb.append(" ");
             coordFormatter.format(z, sb);
         }
+        if (!Double.isNaN(m) && encodeCoordinatesMeasures) {
```

Comment:
It is not that unusually to have coordinates where the measure (say time) is not available on every coordinate. Does GML expect us to write out NaN in these cases?

Reply:
I don't think GML expects M coordinates at all, I just keep the existing behavior, as I say in the comment above, points will be now encode using the coordinates sequence method, so NaN will be add if no value is available.